### PR TITLE
Fix hash types documentation & spec links

### LINKED ISSUE #5388: Restore accidentally removed documentation
### Summary

https://github.com/input-output-hk/plutus/pull/4833 Accidentally removed documentation of `ScriptHash` and `DatumHash` and replaced it with **incorrect** one

### Steps to reproduce the behavior

See https://github.com/input-output-hk/plutus/blob/master/plutus-ledger-api/src/PlutusLedgerApi/V1/Scripts.hs#L78-L79

See https://github.com/input-output-hk/plutus/blob/master/plutus-ledger-api/src/PlutusLedgerApi/V1/Scripts.hs#L90-L91

### Actual Result

N/A

### Expected Result

Docs are correct

### Describe the approach you would take to fix this

_No response_

### System info

Not relevant

### PR DESCRIPTION
**Pull request by t4ccer**
Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Changelog fragments have been written (if appropriate)
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, PNG optimization, etc. are updated
- PR
    - [x] (For external contributions) Corresponding issue exists and is linked in the description
    - [x] Targeting master unless this is a cherry-pick backport
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] Reviewer requested

Close #5388

### CODE REVIEW

**Review by effectfully — APPROVED** [review]
Thanks a lot for catching. Let me see if I can merge the PR.

File: plutus-ledger-api/src/PlutusLedgerApi/V1/Crypto.hs [hunk 1]
```diff
@@ -23,7 +23,7 @@ rather than the key itself. Hashed with /BLAKE2b-224/. 28 bytes.
 
 This is a simple type without any validation, __use with caution__.
 You may want to add checks for its invariants. See the
- [Shelley ledger specification](https://hydra.iohk.io/build/16861845/download/1/ledger-spec.pdf).
+ [Shelley ledger specification](https://github.com/input-output-hk/cardano-ledger/releases/latest/download/shelley-ledger.pdf).
```
[the remark this answers is not in the document]
Comment (t4ccer) [reply plutus-ledger-api/src/PlutusLedgerApi/V1/Crypto.hs:26]: Yeah I can replace it with https://github.com/input-output-hk/cardano-ledger/releases/download/cardano-ledger-spec-2023-04-03/shelley-ledger.pdf, I just took it from `cardano-ledger` readme

File: plutus-ledger-api/src/PlutusLedgerApi/V1/Crypto.hs
[the remark this answers is not in the document]
Comment (t4ccer) [reply plutus-ledger-api/src/PlutusLedgerApi/V1/Crypto.hs:1]: Done!

**Review by effectfully — APPROVED** [review]
Thank you again, it's great to have this fixed.

**Pull request merged**